pallets/ethereum-transaction/src/lib.rs

```diff
@@ -0,0 +1,206 @@
+// Copyright 2023 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use cfg_traits::ethereum::EthereumTransactor;
+use ethereum::{LegacyTransaction, TransactionAction, TransactionSignature, TransactionV2};
+use fp_evm::CallOrCreateInfo;
+use frame_support::{
+	dispatch::{DispatchErrorWithPostInfo, PostDispatchInfo},
+	pallet_prelude::*,
+};
+pub use pallet::*;
+use pallet_evm::{ExitError, ExitFatal, ExitReason};
+use sp_core::{H160, H256, U256};
+use sp_std::vec::Vec;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+const TRANSACTION_RECOVERY_ID: u64 = 42;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub (super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + pallet_ethereum::Config {
+		/// The event type.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+	}
+
+	/// Storage for nonce.
+	#[pallet::storage]
+	pub(crate) type Nonce<T: Config> = StorageValue<_, U256, ValueQuery>;
```

Comment:
Safety first: Is it intentional to have a global nonce without finer granularity such as per account?

Reply:
Yes, since there is no account that is executing the transaction and the signature is fake, as per:

https://centrifuge.hackmd.io/yd0ak2uKQYy92I4bR8ViDA#Existing-Work